### ISSUE TITLE
[FEATURE REQUEST]  One click option in settings to say "Keep existing cookies and auto-delete future cookies"

### ISSUE THREAD

**Issue by bmrz2019** [mrdokenny/Cookie-AutoDelete#1204]
### Acknowledgements

- [X] I acknowledge that I have read the above items

### Is your feature request related to a problem? Please describe.

Instead of manually whitelisting new domains etc. it would nice to say - let the current state of browser be assumed as the 'standard - with some cookies'  - and say delete all future cookies.

Use cases:

1. google sync (user logs into google for passwords/bookmarks/etc) - does not want to logout
2. In my case I log into google, github, gitlab, <couple of work domains> - I want this as default environment. 

(yes, I can always use incognito... but as you know many sites do not work with incognito or I am stuck with never-ending ReCaptcha/spiral galaxy).

### Describe the solution that you'd like

One tick button:

Keep existing cookies and auto-delete future cookies

**Comment by kennethtran93** [comment]
So to clarify, you want a function that checks all existing cookies and create whitelisting for them, specifically checking that cookie to be saved and clean all others.

This will not be a one-click settings option, but if we proceed would be a button in the List of Expressions instead.

For most people this is usually to spend some time to do the initial setup, and use the popup to assist with creating future expressions as new sites are visited.

There's up and downs with this.
1. White/grey lists for unwanted 3rd-party domains may be created.
2. If you're explicitly wanting to only keep specific cookies that are existing, then down the line the site in question may rename/replace/add cookies such that you'd need to make sure these are included/excluded as necessary.
    - If all you want is to simply whitelist all cookies from that specific domain then that is much easier on our end to try to implement I foresee.
3. If there are existing expressions for a domain (including matching wildcard domains), we would not add/override it.

And to respond to your use cases:
1. We're working on browser sync for the extension.  It's currently a one-man dev (with limited free time) with some input designs and review by others.
2. For GitHub and GitLab, you can most likely just whitelist their main domain as the expression, and any other 3rd-party sites if you have used them for logging in.  As for Google...it's nested somewhere in their subdomains for authentications and whatnot.

Have you thought of using Profiles to achieve container separation - i.e. have one Profile just for Google/GitHub/GitLab stuff?

If this request gets more up votes, I would think about it, but until then this will be treated as low priority.

**Comment by bmrz2019** [comment]
> Have you thought of using Profiles to achieve container separation - i.e. have one Profile just for

Yes I do that. But the average user does not. Even for me there are some inconveniences with that. 

My current mode is: I have a folder profile with passwords/cookies for couple of sites. Everytime I start browser I 

```
rsync -a stored_profile_google temp_profile_dir
/usr/bin/chromium-browser --user-data-dir=temp_profile_dir
```

Now all external site tracking is lost.

> White/grey lists for unwanted 3rd-party domains may be created.

Agree.

> may rename/replace/add cookies such that you'd need to make sure these are in

Good point never thought of that. Despite this I think of it like 'snapshot' my browser at this state and clear everything would be OK. Lets say **rename/replace/add cookies** happens - the user can just temporarily 'enable write' for cookies then allow new cookies - then just make cookies 'disabled'.

> unwanted 3rd-party domains may be created.

agree.

Think of this as a making the filesystem read-only after opening browser....


> It's currently a one-man dev 

Your efforts are amazing. Thanks. I shall not demand too much from opensource devs. Too many get burned by work load.

**Comment by kennethtran93** [comment]
If you're rsyncing, wouldn't the extension preferences be synced as well?  Then you would just need to configure it once.  Overall, I think implementing a 'denylist' first may help.  That or the browser sync for the extension.

**Comment by bmrz2019** [comment]
> If you're rsyncing, 


At the moment I do not use the extension.
